ArchitectureIdea/ContentView.swift

```diff
@@ -29,21 +31,24 @@ struct ContentView: View {
                         await model.clearDevices()
                     }
                 }
-                List(model.devices, id: \.self) { device in
+                List(model.recordingState.devices, id: \.self) { device in
                     Text(device)
                 }
             }
         }
         .padding()
+        .animation(.default, value: contentViewModel.recordingState.devices)
```

Comment:
This makes the list of devices animate the changes. One quick way of knowing if everything is working as it should on SwiftUI is that animations work correctly and don't randomly glitch. 

https://www.objc.io/blog/2021/08/24/conditional-view-modifiers/